u8timeseries/models/theta.py

```diff
@@ -0,0 +1,95 @@
+import statsmodels.tsa.holtwinters as hw
+from ..timeseries import TimeSeries
+from .autoregressive_model import AutoRegressiveModel
+import numpy as np
+import math
+from .statistics import check_seasonality, extract_trend_and_seasonality, remove_seasonality
+
+
+class Theta(AutoRegressiveModel):
+    """
+    An implementation of the Theta method with variable value of the `theta` parameter.
+
+    :param theta: User-defined value for the theta parameter. Default to 0.
+
+    TODO: Maybe? Explain a bit what it is?
+    TODO: Implement OTM: Optimized Theta Method (https://arxiv.org/pdf/1503.03529.pdf)
+    TODO: From the OTM, set theta_2 = 2-theta_1 to recover our generalization - but we have an explicit formula.
+    TODO: Try with something different than SES? They do that in the paper.
+    """
+    def __init__(self, theta: int = 0):
+        super().__init__()
+
+        self.model = None
+        self.coef = 1
+        self.alpha = 1
+        self.length = 0
+        self.theta = theta
+        self.is_seasonal = False
+        self.seasonality = None
+        self.season_period = 0
+
+        # The restriction of values is due to the following facts:
+        # - if theta = 1, then the theta method restricts to a simple exponential smoothing (SES)
+        # - if theta = 2, the formula for self.coef below fails.
+
+        assert self.theta not in [1, 2], 'Theta should be neither 1 nor 2.'
+
+    def fit(self, ts, season_period: int = None):
+        """
+        Fits the Theta method with parameter `theta` to the TimeSeries `ts` assuming that \
+                                                    the data has seasonality `season_period`.
+
+
+
+        :param ts: The TimeSeries to fit.
+        :param season_period: User-defined seasonality period. Default to None.
+        """
+        super().fit(ts)
+
+        self.length = len(ts)
+
+        # Check for statistical significance of user-defined season period
+        # or infers season_period from the TimeSeries itself.
+        self.is_seasonal, self.season_period = check_seasonality(ts, season_period)
+
+        # Store and remove seasonality effect if there is any.
```

Comment:
You should probably initate `new_ts` before this `if`, otherwise `new_ts` won't be defined on line 62 if the user calls the function with a "wrong" seasonality